osu.Game/Screens/Edit/EditorBeatmap.cs

```diff
@@ -69,7 +69,8 @@ public class EditorBeatmap : TransactionalCommitComponent, IBeatmap, IBeatSnapPr
         public EditorBeatmap(IBeatmap playableBeatmap, ISkin beatmapSkin = null)
         {
             PlayableBeatmap = playableBeatmap;
-            BeatmapSkin = beatmapSkin;
+            if (beatmapSkin is Skin skin)
```

Comment:
Is there an expected case of `beatmapSkin` to be non-null but not a `Skin`? If not, the type of `beatmapSkin` could be changed to `Skin`.

Reply:
At which point would you expect the cast to `Skin`? I think it still needs to happen somewhere, since we probably want to keep the `ISkin` specification on `WorkingBeatmap.GetSkin()` for flexibility.

Reply:
I was thinking about explicitly casting `ISkin` to `Skin` in the usage site.
If a `Skin` is expected for this class, I think it makes sense to let the usage sites to decide how to handle, rather than silently ignoring invalid cases.

Reply:
I've be okay with a direct cast with some kind of `Debug.Assert` guarding it, at the parent site I guess?

Reply:
A side note here, though: the beatmap skin may be null here. So a degree of tolerance for that is still probably required.